src/views/SettingsView.vue

```diff
@@ -1,31 +1,31 @@
 <template>
   <div>
     <div class="ui container">
-      <h2>Settings</h2>
+      <h2>{{$t('label.settings')}}</h2>
       <div class="ui action input labeled">
-        <div class="ui label">Language</div>
-        <input placeholder="Language" v-model="lang" />
-        <button class="ui button primary" @click="saveLang">Save</button>
+        <div class="ui label">{{$t('label.language')}}</div>
+        <select v-model="$i18n.locale">
+          <option v-for="(lang, i) in langs" :key="`Lang${i}`" :value="lang">{{ lang }}</option>
+        </select>
+        <button class="ui button primary" @click="saveLang">{{$t('label.save')}}</button>
       </div>
     </div>
   </div>
 </template>
 
 <script>
 import { localSettings } from "../settings";
+import messages from '../i18n/messages'
 
 export default {
   name: "SettingsView",
   props: [],
   data: function() {
-    const settings = localSettings.fetch();
-    return {
-      lang: settings.lang || "en"
-    };
+    return { langs: Object.keys(messages) }
   },
   methods: {
     saveLang: function() {
-      localSettings.update("lang", this.lang);
+      localSettings.update("lang", this.$i18n.locale);
```

Comment:
The language is updated as soon as we switch it, better to wait for a click on the save button